extensions/vertx/runtime/src/main/java/io/quarkus/vertx/VertxContextSupport.java

```diff
@@ -96,14 +97,44 @@ public void accept(MultiSubscribe<T> ms) {
         });
     }
 
-    private static Context getContext() {
+    /**
+     * Executes the supplied blocking {@link Callable} on a Vertx duplicated context; does not block the current thread.
+     * <p>
+     * If necessary, the CDI request context is activated during execution of the blocking code.
+     *
+     * @param <T>
+     * @param callable
+     * @return the produced {@link Uni}
+     * @see VertxContext#getOrCreateDuplicatedContext(Vertx)
+     */
+    public static <T> Uni<T> executeBlocking(Callable<T> callable) {
+        Context context = getContext(true);
+        return Uni.createFrom().completionStage(() -> {
```

Comment:
defer might be slightly better.

@jponge wdyt?

Reply:
> defer might be slightly better.

In what exactly? I mean, both suppliers are called at subscription time. And we would need to convert the `io.vertx.core.Future` to the `Uni` anyway.

Reply:
I was referring to https://javadoc.io/doc/io.smallrye.reactive/mutiny/latest/io.smallrye.mutiny/io/smallrye/mutiny/groups/UniCreate.html which avoid using completion stages and it's slightly more optimized. But, yes, then you need an emitter, so the cost would be almost equivalent.

Reply:
> I was referring to https://javadoc.io/doc/io.smallrye.reactive/mutiny/latest/io.smallrye.mutiny/io/smallrye/mutiny/groups/UniCreate.html which avoid using completion stages and it's slightly more optimized.

Did you mean https://javadoc.io/doc/io.smallrye.reactive/mutiny/latest/io.smallrye.mutiny/io/smallrye/mutiny/groups/UniCreate.html#deferred(java.util.function.Supplier)?

Reply:
Yes, sorry, missed the anchor.

Reply:
Deferred with supplier is better, but a `CompletionStage` isn't incorrect per-se.

Reply:
> Deferred with supplier is better, but a `CompletionStage` isn't incorrect per-se.

Better in terms of performance, or?

Hm, but how do I create a `Uni` from `io.vertx.core.Future` without `Uni.createFrom().completionStage()`? That's what `io.smallrye.mutiny.vertx.UniHelper.toUni(Future<T>)` does by the way.

Reply:
Yes yes you are right